src/setup/node.rs

```diff
@@ -0,0 +1,167 @@
+//! High level APIs and types for node setup and teardown.
+
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+};
+
+use anyhow::Result;
+use fs_extra::dir;
+
+use super::constants::{ALGORAND_SETUP_DIR, NODE_DIR, PRIVATE_NETWORK_DIR};
+use crate::setup::{
+    config::{NodeConfig, NodeMetaData},
+    get_algorand_work_path,
+};
+
+pub enum ChildExitCode {
+    Success,
+    ErrorCode(Option<i32>),
+}
+
+pub struct NodeBuilder {
+    /// Node's startup configuration.
+    conf: NodeConfig,
+    /// Node's process metadata read from Ziggurat configuration files.
+    meta: NodeMetaData,
+}
+
+impl NodeBuilder {
+    /// Creates a new [NodeBuilder].
+    pub fn new() -> anyhow::Result<Self> {
+        let setup_path = get_algorand_work_path()?.join(ALGORAND_SETUP_DIR);
+
+        let conf = NodeConfig::default();
+        let meta = NodeMetaData::new(setup_path)?;
+
+        Ok(Self { conf, meta })
+    }
+
+    /// Creates a [Node] according to configuration and starts its process.
+    pub async fn start(&mut self, target: &Path) -> Result<Node> {
```

Comment:
```suggestion
    pub fn start(&mut self, target: &Path) -> Result<Node> {
```
I see no `await` underneath :detective: